wdl/LRVisualizeVariants.wdl

```diff
@@ -0,0 +1,172 @@
+            beds = VCFToBed.bed,
+            ref_fai = ref_map['fai'],
+            slop = 50000,
+            dist = 50000,
+            prefix = "union"
+    }
+
+    # Parallelize read extractions
+    scatter (n in range(num_simultaneous_jobs)) {
+        call ExtractLoci {
+            input:
+                aligned_bams     = aligned_bams,
+                aligned_bais     = aligned_bais,
+                bed              = MergeBedFiles.merged_bed,
+                nth              = n,
+                num_jobs         = num_simultaneous_jobs,
+                gcs_out_root_dir = outdir
+        }
+    }
+
+    ##########
+    # return the path to the visualization directory
+    ##########
+
+    output {
+        String viz_path = outdir
+    }
+}
+
+task ExtractLoci {
+    input {
+        Array[String] aligned_bams
+        Array[String] aligned_bais
+
+        File bed
+        Int nth
+        Int num_jobs
+
+        String gcs_out_root_dir
+
+        RuntimeAttr? runtime_attr_override
+    }
+
+    # estimate required disk size in GB
+    Int disk_size = 10 + ceil(0.3*length(aligned_bams))
+
+    command <<<
+        # Authorize streaming access to GCS data
+        source /opt/re-auth.sh
+        set -euxo pipefail
+
+        RET=0
+
+        # Iterate over every nth line in the bed file
+        awk 'NR % ~{num_jobs} == ~{nth} { print $0 }' ~{bed} | while read line; do
+            # Extract locus pieces
+            chr=$(echo $line | awk '{ print $1 }')
+            start=$(echo $line | awk '{ print $2 }')
+            stop=$(echo $line | awk '{ print $3 }')
+
+            # Set up local and remote BAM file paths
+            local_bam_file="viz.${chr}_${start}_${stop}.bam"
+            local_bai_file="viz.${chr}_${start}_${stop}.bam.bai"
+            gcs_bam_file="~{gcs_out_root_dir}/$local_bam_file"
+            gcs_bai_file="~{gcs_out_root_dir}/$local_bai_file"
+
+            if gsutil -q stat "$gcs_bam_file" ; then
+                # Skip processing BAMs that we've already written to GCS
+                echo "$gcs_bam_file already exists."
+            else
+                # Remove any lingering subset BAMs
+                rm -f *.subset.bam*
+
+                # Loop over all sample BAMs
+                while read sample_bam; do
+                    # Reauthorize streaming access to GCS with each iteration
+                    source /opt/re-auth.sh
+
+                    sample_bai=$(echo $sample_bam | sed 's/.bam/.bam.bai/')
+                    name=$(basename $sample_bam .bam)
+
+                    # Extract locus from sample BAM
+                    samtools view \
+                        -bhX \
+                        -M \
+                        -@ 1 \
+                        --verbosity=8 \
+                        --write-index \
+                        -o "${name}.subset.bam##idx##${name}.subset.bai" \
+                        $sample_bam $sample_bai \
+                        ${chr}:${start}-${stop}
```

Comment:
One small thing: this uses 1-based system, while the start and stop were extracted from a 0-based system.